lib/tasks_private/instance_types.rake

```diff
@@ -1,215 +1,191 @@
-      :name                    => "#{instance[:name]}",
-      :family                  => "#{instance[:family]}",
-      :description             => "#{instance[:description]}",
-      :memory                  => #{instance[:memory] / Numeric::GIGABYTE}.gigabytes,
-      :vcpu                    => #{instance[:vcpu]},
-      :ebs_only                => #{instance[:ebs_only]},
-      :instance_store_size     => #{instance[:instance_store_size] > 0 ? (instance[:instance_store_size] / Numeric::GIGABYTE).to_s + '.gigabytes' : 0},
-      :instance_store_volumes  => #{instance[:instance_store_volumes]},#{instance[:instance_store_ssd] ? ' # SSD' : ''}
-      :architecture            => #{instance[:architecture]},
-      :virtualization_type     => #{instance[:virtualization_type]},
-      :network_performance     => :#{instance[:network_performance]},
-      :physical_processor      => "#{instance[:physical_processor]}",
-      :processor_clock_speed   => "#{instance[:processor_clock_speed]}",
-      :intel_aes_ni            => #{instance[:intel_aes_ni] ? 'true' : 'nil'},
-      :intel_avx               => #{instance[:intel_avx] ? 'true' : 'nil'},
-      :intel_avx2              => #{instance[:intel_avx2] ? 'true' : 'nil'},
-      :intel_turbo             => #{instance[:intel_turbo] ? 'true' : 'nil'},
-      :ebs_optimized_available => #{instance[:ebs_optimized_available] ? 'true' : 'nil'},
-      :enhanced_networking     => #{instance[:enhanced_networking] ? 'true' : 'nil'},
-      :cluster_networking      => #{instance[:cluster_networking]  ? 'true' : 'nil'},
-      :vpc_only                => #{instance[:vpc_only]},
-    },
-    EOT
-  end
+# frozen_string_literal: true
 
-  def cluster_networking?(type)
-    # https://aws.amazon.com/ec2/instance-types/#Cluster_Networking
-    type.split('.').first.in?(%w(r4 x1 m4 c4 c3 i2 cr1 g2 hs1 p2 d2))
-  end
+begin
+  require 'awesome_print'
+rescue LoadError
+end
+
+namespace 'aws:extract' do
+  desc 'Get / renew instance types and details list from AWS Price List Bulk API'
+  task :instance_types do
+    if (GH_API_TOKEN = ENV['GITHUB_API_TOKEN']).blank?
+      raise ArgumentError, 'No GitHub API key found in ENV'
+    end
+
+    require_relative 'lib/aws_products_data_collector'
+    require_relative 'lib/aws_instance_data_parser'
+
+    data_dir = ManageIQ::Providers::Amazon::Engine.root.join('db/fixtures')
+    data_dir.mkpath
+    out_file = data_dir.join('aws_instance_types.yml')
 
-  def format_instance_miq(i)
-    {}.tap do |o|
-      # x1.16large manually fixed to x1.16xlarge
-      # i3.16large manually fixed to i3.16xlarge
-      %i(family instance_store_size instance_store_volumes vpc_only enhanced_networking).each do |key|
-        o[key] = i[key]
+    # weird cache logging issue workaround
+    I18n.backend = I18n.backend.backend unless Rails.initialized?
```

Comment:
Really curious about this...never seen this before...if you found it via the internet can you include a link to it in the comments for future readers?

Reply:
No, I found the solution with debugging.
Without: `I18n::InvalidLocale: :en is not a valid locale`